provider/azure/doc.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package azure implements the Azure provider, registered with
+// environs under the name "azure". The provider implements the
+// [github.com/juju/juju/environs.Environ] interface, which defines
+// methods for provisioning compute, network, and storage resources.
+//
+// Here we describe some key implementation details specific to the Azure provider.
+//
+// # SDK
+//
+// The provider implementation is built using the [Azure SDK].
+//
+// # Credential Types
+//
+// The supported credential types are:
+//   - service-principal-secret
+//   - managed-identity
+//
+// The recommended way to create a credential with which to bootstrap is
+// to run
+//
+//	juju add-credential azure
+//
+// and follow the prompts. Choosing "interactive" is the best choice for
+// setting up a service principal credential as the browser is used to
+// log into an Azure account and the credential attributes are filled
+// in automatically.
+//
+// # Resource Groups
+//
+// All models, including the controller model, are created in a resource group.
+// The resource group contains all artefacts for the model, including:
+//   - virtual machines
+//   - disks
+//   - networks and subnets
+//   - security groups and rules
+//   - public IP addresses
+//   - availability sets
+//   - key vaults
+//
+// # Provisioning Resources
+//
+// During bootstrap, a deployment API client is used to deploy an [Azure Resource Manager]
+// template which contains all compute, network, and storage resources for the controller.
+// After bootstrap, API clients are created for particular resource types defined in the SDK.
+// All API clients use the same options which define the retry and logging behaviour.
+//
+// # Availability Sets
```

Comment:
Based on our discussion, this is an upstream feature, not part of the Juju mould for providers. The current structure is based on the mould. So maybe this one can become "Upstream features currently not implemeted in this provider".

Reply:
The provider **does** know about availability sets because it configures the deployment to use them. So they are implemented in this provider by definition. The reason for calling this info out is that Azure availability sets have somewhat different semantics compared with availability zones used by other providers. I've reworked the text and heading to hopefully explain a bit better.